core/src/main/scala/org/apache/spark/HttpServer.scala

```diff
@@ -155,6 +158,7 @@ private[spark] class HttpServer(
       throw new ServerStateException("Server is already stopped")
     } else {
       server.stop()
+      condOpt(server.getThreadPool) { case x: LifeCycle => x }.foreach(_.stop())
```

Comment:
Yes it is effectively doing the same thing, but using the `condOpt` from `PartitialFunction` instead. I can change it to your version if it is more preferable.